src/storage/sftp/storage.c

```diff
@@ -37,6 +38,199 @@ struct StorageSftp
     TimeMSec timeout;                                               // Session timeout
 };
 
+/***********************************************************************************************************************************
+Return known host key type based on host key type. LIBSSH_ constants defined in libssh2.h
+***********************************************************************************************************************************/
+static int
+storageSftpKnownHostKeyType(const int hostKeyType)
+{
+    FUNCTION_TEST_BEGIN();
+        FUNCTION_TEST_PARAM(INT, hostKeyType);
+    FUNCTION_TEST_END();
+
+    int knownHostKeyType;
+
+    switch (hostKeyType)
+    {
+        case LIBSSH2_HOSTKEY_TYPE_RSA:
+            knownHostKeyType = LIBSSH2_KNOWNHOST_KEY_SSHRSA;
+            break;
+
+        case LIBSSH2_HOSTKEY_TYPE_DSS:
+            knownHostKeyType = LIBSSH2_KNOWNHOST_KEY_SSHDSS;
+            break;
+
+#ifdef LIBSSH2_HOSTKEY_TYPE_ECDSA_256
+        case LIBSSH2_HOSTKEY_TYPE_ECDSA_256:
+            knownHostKeyType = LIBSSH2_KNOWNHOST_KEY_ECDSA_256;
+            break;
+#endif
+
+#ifdef LIBSSH2_HOSTKEY_TYPE_ECDSA_384
+        case LIBSSH2_HOSTKEY_TYPE_ECDSA_384:
+            knownHostKeyType = LIBSSH2_KNOWNHOST_KEY_ECDSA_384;
+            break;
+#endif
+
+#ifdef LIBSSH2_HOSTKEY_TYPE_ECDSA_521
+        case LIBSSH2_HOSTKEY_TYPE_ECDSA_521:
+            knownHostKeyType = LIBSSH2_KNOWNHOST_KEY_ECDSA_521;
+            break;
+#endif
+
+#ifdef LIBSSH2_HOSTKEY_TYPE_ED25519
+        case LIBSSH2_HOSTKEY_TYPE_ED25519:
+            knownHostKeyType = LIBSSH2_KNOWNHOST_KEY_ED25519;
+            break;
+#endif
+
+        default:
+            knownHostKeyType = 0;
+            break;
+    }
+
+    FUNCTION_TEST_RETURN(INT, knownHostKeyType);
+}
+
+/***********************************************************************************************************************************
+Return a match failed message based on known host check failure type
+***********************************************************************************************************************************/
+static const char *
+storageSftpKnownHostCheckpFailureMsg(const int rc)
+{
+    FUNCTION_TEST_BEGIN();
+        FUNCTION_TEST_PARAM(INT, rc);
+    FUNCTION_TEST_END();
+
+    const char *matchFailMsg;
+
+    switch (rc)
+    {
+        case LIBSSH2_KNOWNHOST_CHECK_FAILURE:
+            matchFailMsg = "LIBSSH2_KNOWNHOST_CHECK_FAILURE";
+            break;
+
+        case LIBSSH2_KNOWNHOST_CHECK_NOTFOUND:
+            matchFailMsg = "not found in known hosts files: LIBSSH2_KNOWNHOST_CHECK_NOTFOUND";
+            break;
+
+        case LIBSSH2_KNOWNHOST_CHECK_MISMATCH:
+            matchFailMsg = "mismatch in known hosts files: LIBSSH2_KNOWNHOST_CHECK_MISMATCH";
+            break;
+
+        default:
+            matchFailMsg = "experienced an unknown failure:";
```

Comment:
How about "unknown failure" here? Not sure we need the : at the end (it doesn't really fit with the eventual formatting).

Reply:
Updates in
https://github.com/pgbackrest/pgbackrest/pull/2125/commits/4fd017c80716d71a7ffff1f25379b9da196f7001
https://github.com/pgbackrest/pgbackrest/pull/2125/commits/6811405f150c8f1095c3da31d952f0ddebc58be1
https://github.com/pgbackrest/pgbackrest/pull/2125/commits/a3a82046feab0a915e0c116f867818b86ae42621